### PR TITLE
Improving k8s 1.25 compatibility for rancher/chart

### PR DESCRIPTION
**Pull request by MbolotSuse**
## Issue: 
#41455
 
## Problem
The current default for `global.cattle.psp.enabled` is not compatible with k8s 1.25. This requires users to manually set this value to `false` when installing on this version of k8s. As 1.25+ is becoming more widely adapted, this pain point will become more apparent for our users.
 
## Solution
- There is no default for `global.cattle.psp.enabled` (i.e. the default value is "").
  - If this value is "" and the cluster does not support PSPs (i.e. is k8s version 1.25 or higher), then PSPs will not be used.
  - If this value is "" and the cluster does support PSPs (i.e. is k8s version 1.24 or lower), then PSPs will be used.
- If the value is set to true, no capabilities check will be used, and the chart will attempt to use PSPs. This will cause the above error message for k8s 1.25 clusters.
- If the value is set to false, no capabilities check will be used, and the chart will not attempt to use PSPs.
 
 The changes to the chart caused our charts tests to fail, since it used the `lookup` function (to determine if there is an active k8s cluster this is running against), which our old fork doesn't appear to support. To resolve this, I refactored our tests and our CI to use the upstream [unittest plugin](https://github.com/helm-unittest/helm-unittest)
 
## Testing
Repro steps are valid.

### Manual Testing
Basic `helm template` and `helm install` cases were done, but more will need to be re-tested here.

### Automated Testing
I added tests for the chart when `global.cattle.psp.enabled` is set to:
- True
- False
- Unset
- Set to an invalid value

These tests do not cover the "live cluster cases" since the `helm-unittest` framework does not appear to offer the ability to mock out the `lookup` functionality.

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -49,15 +47,19 @@ RUN curl -sLf ${!HELM_URL_V2} -o /usr/bin/rancher-helm && \
...
+    helm init -c --stable-repo-url https://charts.helm.sh/stable/
 
 # set up helm 3
 RUN mkdir /usr/tmp && \
     curl ${HELM_URL_V3} | tar xvzf - --strip-components=1 -C /usr/tmp/ && \
     mv /usr/tmp/helm /usr/bin/helm_v3 && \
     chmod +x /usr/bin/kustomize
 
+# set up helm unittest - doesn't need to run for s390x architecture
+RUN if [ "${ARCH}" == "arm64" || "${ARCH}" == "amd64" ]; then \
```
Comment (superseb) [inline Dockerfile.dapper:59]: Can be just one arch if there is no benefit on multi arch

File: Dockerfile.dapper [hunk 2]
```diff
@@ -49,15 +47,19 @@ RUN curl -sLf ${!HELM_URL_V2} -o /usr/bin/rancher-helm && \
...
+        helm plugin install https://github.com/helm-unittest/helm-unittest.git --version 0.3.2 \
```
Comment (superseb) [inline Dockerfile.dapper:60]: This version still needs to be in an environment variable for easy (automated) updating

**Comment by superseb** [comment]
I believe we should move to `helm_v3` for this so we use the latest and supported version of Helm instead of Helm 2

**Review by superseb — APPROVED** [review]

**Review by rmweir — APPROVED** [review]
LGTM

**Pull request merged**